pydrive2/fs/spec.py

```diff
@@ -542,6 +542,43 @@ def cp_file(self, lpath, rpath, **kwargs):
             buffer = io.BytesIO(stream.read())
             self.upload_fobj(buffer, rpath)
 
+    @_gdrive_retry
+    def mv(self, path1, path2, recursive=False, maxdepth=None, **kwargs):
+
+        if maxdepth:
+            raise NotImplementedError("Max depth move is not supported")
+
+        if self.isdir(path1) and not recursive:
+            raise AttributeError(
+                "Cannot move a directory without recursive flag"
+            )
+
+        file1_name = posixpath.basename(path1)
+
+        file1_parent = self._parent(path1)
+
+        if self.exists(path2):
+            file2_name = file1_name
+            file2_parent = path2
+        else:
+            file2_name = posixpath.basename(path2)
+            file2_parent = self._parent(path2)
+
+        file1_id = self._get_item_id(path1)
+
+        file1 = self.client.CreateFile({"id": file1_id})
+
+        if file1_name != file2_name:
+            file1["title"] = file2_name
+
+        if file1_parent != file2_parent:
+            file2_parent_id = self._get_item_id(file2_parent)
+            file1["parents"] = [{"id": file2_parent_id}]
+
+        file1.Upload()
+
+        # TODO need to invalidate the cache for the old path
```

Comment:
Let's move it up, before the call, create a ticket in GH and put a link to it pls

Reply:
I created the issue #232 and referenced it in the comment.

Reply:
thanks!